pkg/kubernetes/kubernetes.go

```diff
@@ -188,32 +188,21 @@ func resolveConfig(ctx string) *rest.Config {
 	return internal
 }
 
-func (k Kubernetes) GetNodePort(service *v1.Service) int32 {
+func (k Kubernetes) GetNodePort(service *corev1.Service) int32 {
 	return service.Spec.Ports[0].NodePort
 }
 
-// PublicIP returns the public IP address of the Kubernetes cluster
-func (k Kubernetes) PublicIP() string {
-	u, _ := url.Parse(k.RestConfig.Host)
-	return u.Hostname()
-}
-
 // GetNodesHost return the addresses of the k8s nodes
 func (k Kubernetes) GetNodesHost() ([]string, error) {
```

Comment:
I wondering if we can get rid of this since it's used only here: https://github.com/infinispan/infinispan-operator/blob/7ebaad9a1eca9761709fa6253a96e3b67d4b1a61/test/e2e/utils/kubernetes.go#L319-L320

Reply:
Yeah, good point, will try to move to the test package

Reply:
maybe it can be just replaced with a GetNodeHost, the test just pickup the first result

Reply:
Switch to the `k.Kubernetes.GetNodeHost`. Seems to be logic a bit similar with unnecessary variations